main/main.go

```diff
@@ -431,20 +430,26 @@ func createApplicationRelayersForSourceChain(
 	// The Listener begins processing messages starting from the minimum height across all the ApplicationRelayers
 	minHeight := uint64(0)
 	for _, relayerID := range database.GetSourceBlockchainRelayerIDs(&sourceBlockchain) {
-		height, err := database.CalculateStartingBlockHeight(
-			logger,
-			db,
-			relayerID,
-			sourceBlockchain.ProcessHistoricalBlocksFromHeight,
-			currentHeight,
-		)
-		if err != nil {
-			logger.Error(
-				"Failed to calculate starting block height",
-				zap.String("relayerID", relayerID.ID.String()),
-				zap.Error(err),
+		// Calculate the starting block height for the relayer only if we're processing historical blocks
+		// Otherwies, the first block we process is the next block after the current height
+		height := currentHeight + 1
```

Comment:
nit, free to drop: I  prefer having this in an else statement below the `if cfg.ProcessMissedBlocks` instead of setting it here and overwriting it.

Reply:
Yeah, I agree the else statement would make it clearer

Reply:
I went with a slightly different approach so that we can emit the `process-missed-blocks=false` log only once per source chain. Lmk if you agree.